### PR TITLE
Separate cache data sources into basic and extended

### PR DESCRIPTION
**Pull request by oldalton**

### CODE REVIEW

**Comment by unpluggedk** [comment]
My take on this would be that 
1. appMetadata goes into metadata cache,and
2. metadata cache becomes a part of extended.

File: IdentityCore/src/MSIDConstants.m [hunk 1]
```diff
@@ -56,3 +56,6 @@
 NSString *const MSIDTrustedAuthorityWorldWide    = @"login.microsoftonline.com";
 NSString *const MSIDTrustedAuthorityUSGovernment = @"login-us.microsoftonline.com";
 NSString *const MSIDTrustedAuthorityCloudGovApi  = @"login.usgovcloudapi.net";
+
+NSString *const MSID_DEFAULT_AAD_AUTHORITY       = @"https://login.microsoftonline.com/common";
+NSString *const MSID_DEFAULT_MSA_TENANTID        = @"9188040d-6c67-4c5b-b112-36a304b66dad";
```
Comment (Om83) [inline IdentityCore/src/MSIDConstants.m]: If this is the MegaTenant Id, would be good to call it the MegaTenant.

**Comment by Om83** [comment]
Description is missing. Would be good to call out the the different challenges of the existing cache design and what we plan to achieve.

File: IdentityCore/src/MSIDConstants.m (hunk 1 above)
Comment (oldalton) [reply IdentityCore/src/MSIDConstants.m]: It's not known as such externally as far as I know.
Here's the documentation: 
A GUID that represents the Azure AD tenant that the user is from. For work and school accounts, the GUID is the immutable tenant ID of the organization that the user belongs to. For personal accounts, the value is 9188040d-6c67-4c5b-b112-36a304b66dad. 

Leaving as is.

**Comment by oldalton** [comment]
Main challenge is having lots of unimplemented methods because particular cache datasource doesn't support it. Additionally, having to add a separate serializer which essentially does the same for each new cache item type.

File: IdentityCore/src/cache/serializers/MSIDCacheItemJsonSerializer.m [hunk 2]
```diff
@@ -78,16 +70,7 @@ - (NSData *)serializeAccountCacheItem:(MSIDAccountCacheItem *)item
 
 - (MSIDAccountCacheItem *)deserializeAccountCacheItem:(NSData *)data
```
Comment (antrix1989) [inline IdentityCore/src/cache/serializers/MSIDCacheItemJsonSerializer.m]: nit: This one and other similar methods are not used anymore.

Comment (oldalton) [reply IdentityCore/src/cache/serializers/MSIDCacheItemJsonSerializer.m]: Yes! Good catch :)

**Pull request merged**